hail/src/test/scala/is/hail/expr/ir/ArrayFunctionsSuite.scala

```diff
@@ -253,6 +253,45 @@ class ArrayFunctionsSuite extends TestNGSuite {
   @Test(dataProvider = "flatten")
   def flatten(in: IndexedSeq[IndexedSeq[Integer]], expected: IndexedSeq[Int]) {
     assertEvalsTo(invoke("flatten", MakeArray(in.map(toIRArray(_)), TArray(TArray(TInt32())))), expected)
+  }
+
+  @Test def testContains() {
+    val t = TArray(TString())
+
+    assertEvalsTo(
+      invoke("contains", In(0, t), Str("a")),
+      args = FastIndexedSeq(FastIndexedSeq() -> t),
```

Comment:
This only tests the new contains for arrays in IR. Do we need tests of the simplify methods?

Reply:
I'll add something to SimplifySuite, but in general we don't have a good story for isolating a specific rewrite behavior in testing. Running `assertEvalsTo` will compare the result of evaluating an IR without optimization with full optimization, and it's totally possible that other rules will eliminate the specific rewrite we might be interested in testing.